tap_github/__init__.py

```diff
@@ -99,25 +131,40 @@ def get_bookmark(state, repo, stream_name, bookmark_key):
         return repo_stream_dict.get(bookmark_key)
     return None
 
+def raise_for_error(resp):
+    try:
+        resp.raise_for_status()
+    except (requests.HTTPError, requests.ConnectionError) as error:
+        try:
+            error_code = resp.status_code
+            try:
+                response_json = resp.json()
+            except Exception:
+                response_json = {}
+
+            message = "HTTP-error-code: {}, Error: {}".format(
+                error_code, ERROR_CODE_EXCEPTION_MAPPING.get(error_code, {}).get("message", "Unknown Error") if response_json == {} else response_json)
+
+            exc = ERROR_CODE_EXCEPTION_MAPPING.get(error_code, {}).get("raise_exception", GithubException)
+            raise exc(message) from None
+
+        except (ValueError, TypeError):
+            raise GithubException(error) from None
+
 # pylint: disable=dangerous-default-value
 def authed_get(source, url, headers={}):
     with metrics.http_request_timer(source) as timer:
         session.headers.update(headers)
         resp = session.request(method='get', url=url)
```

Comment:
Add raise_for_error() here instead of every function which call authed_get function.

Reply:
@savan-chovatiya Done the changes